### DISCUSSION TITLE
Error in Windows Installation

### DISCUSSION THREAD

**Discussion by cloudlakecho** [ImageMagick/ImageMagick#4853] [Development]
### ImageMagick version

7

### Operating system

Windows

### Operating system, version and so on

Windows 10

### Description

I tried to install ImageMagick-Windows, but I am facing error when I run `CloneRepositories.cmd`

These are my error messages:
```
PS C:\code\from_internet\ImageMagick-Windows> .\CloneRepositories.cmd   
./CloneRepositories.sh: line 2: $'\r': command not found
./CloneRepositories.sh: line 3: syntax error near unexpected token `$'\r''
'/CloneRepositories.sh: line 3: `clone_repository()      
```

When I open the `ClondRepositories.cmd` the line 2 looks empty. Should I remove all empty line in the CMD file? 

### Steps to Reproduce

(1) Clone ImageMgick-Windows  
(2) `$ cd ImageMagick-Windows`  
(3)  `$ ./CloneRepositories.cmd`

**Comment by fmw42** [comment]
Perhaps you should install from the official binary repository. See https://imagemagick.org/script/download.php#windows

**Reply by cloudlakecho** [reply]
@fmw42 Thanks for the suggestion. 
I will try the snibgo's idea first. If his not solve, I will try the binary file.

**Comment by snibgo** [comment]
```
./CloneRepositories.sh: line 2: $'\r': command not found
```
"\\r" is carriage-return. I suppose CloneRepositories.sh has Windows style line-ends (\\r\\n) rather than Unix style (\\n only), and whatever shell you are using expects Unix style.

I suggest you do whatever is needed to tell your shell to accept either style of line endings.

**Reply by cloudlakecho** [reply]
@snibgo Thanks for suggestion. It looks like I have to modify the line ending in the Bash file. 

For the reference, I tried the command line in the both PowerShell and Command Prompt (CMD) in Windows.

**Reply by dlemstra** [reply]
Maybe your git clone settings are set to doing a clone with Linux line endings?

**Reply by cloudlakecho** [reply]
@dlemstra Thanks for suggestion. The "git clone settings" do you mean `git config --list`? I don't see the option. 

I couldn't find to way to replace '/n' to '/r'. 

I downloaded a zip file (instead clone from Git Bash) from [ImageMagick-Windows](https://github.com/ImageMagick/ImageMagick-Windows), but it gave the same error. 

As @fmw42 suggested, I installed the binary version, "[ImageMagick-7.1.0-26-Q16-HDRI-x64-dll.exe](https://download.imagemagick.org/ImageMagick/download/binaries/ImageMagick-7.1.0-26-Q16-HDRI-x64-dll.exe)", but the file also gave an different error at the end of installation. 

I will try to use DLL from the installation in my code and share the outcome, thanks.

**Reply by cloudlakecho** [reply]
I am able to compile and run sample code with the LIB file and the include folder from the binary version. 

I bypass the error, and I may stop trying using the CMD file (instruction). 

This discussion should be opened for other people, right?
